client/compliance_client.go

```diff
@@ -585,6 +654,849 @@ func (c *complianceGRPCClient) CancelOperation(ctx context.Context, req *longrun
 	return err
 }
 
+// RepeatDataBody this method echoes the ComplianceData request. This method exercises
+// sending the entire request object in the REST body.
+func (c *complianceRESTClient) RepeatDataBody(ctx context.Context, req *genprotopb.RepeatRequest, opts ...gax.CallOption) (*genprotopb.RepeatResponse, error) {
+	m := protojson.MarshalOptions{AllowPartial: true, EmitUnpopulated: true}
+	jsonReq, err := m.Marshal(req)
+	if err != nil {
+		return nil, err
+	}
+
+	queryParamStrs := []string{}
+	if req.GetInfo() != nil {
+		queryParamStrs = append(queryParamStrs, fmt.Sprintf("info=%v", req.GetInfo()))
+	}
+	if req.GetName() != "" {
+		queryParamStrs = append(queryParamStrs, fmt.Sprintf("name=%v", req.GetName()))
+	}
+	if req.GetServerVerify() {
+		queryParamStrs = append(queryParamStrs, fmt.Sprintf("serverVerify=%v", req.GetServerVerify()))
+	}
+	query := strings.ReplaceAll(strings.Join(queryParamStrs, "&"), " ", "+")
+
+	url := fmt.Sprintf("%s/v1beta1/repeat:body",
+		c.endpoint,
+	)
+
+	if query != "" {
+		url += "?" + query
+	}
+
+	httpReq, err := http.NewRequestWithContext(ctx, "POST", url, bytes.NewReader(jsonReq))
+	if err != nil {
+		return nil, err
+	}
+	// Set the headers
+	for k, v := range c.xGoogMetadata {
+		httpReq.Header[k] = v
+	}
+	httpReq.Header["Content-Type"] = []string{"application/json"}
+
+	httpRsp, err := c.httpClient.Do(httpReq)
+	if err != nil {
+		return nil, err
+	}
+	defer httpRsp.Body.Close()
+
+	if httpRsp.StatusCode >= http.StatusOK {
+		return nil, fmt.Errorf(httpRsp.Status)
+	}
+
+	buf, err := ioutil.ReadAll(httpRsp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	unm := protojson.UnmarshalOptions{AllowPartial: true, DiscardUnknown: true}
+	rsp := &genprotopb.RepeatResponse{}
+
+	return rsp, unm.Unmarshal(buf, rsp)
+}
+
+// RepeatDataBodyInfo this method echoes the ComplianceData request. This method exercises
+// sending the a message-type field in the REST body. Per AIP-127, only
+// top-level, non-repeated fields can be sent this way.
+func (c *complianceRESTClient) RepeatDataBodyInfo(ctx context.Context, req *genprotopb.RepeatRequest, opts ...gax.CallOption) (*genprotopb.RepeatResponse, error) {
+	m := protojson.MarshalOptions{AllowPartial: true, EmitUnpopulated: true}
+	jsonReq, err := m.Marshal(req)
+	if err != nil {
+		return nil, err
+	}
+
+	body := req.GetInfo()
+	queryParamStrs := []string{}
+	if body.GetFBool() {
```

Comment:
So `RepeatDataBody` has, in it's `google.api.http` annotation, `body: "*"`. Thus, it should have no query params. See https://cloud.google.com/endpoints/docs/grpc-service-config/reference/rpc/google.api#rules-for-http-mapping

Reply:
minor correction: My comment is actually for `RepeatDataBody` (which is the one that has `body: "*"`), i.e. the previous method, which should have no query params.

Here, since all the query params you compute are included inside the `RepeatDataBodyInfo`'s `body: "info"` field, they shouldn't be query params either.

Refer also to https://google.aip.dev/127